### DISCUSSION TITLE
proxyMap doesn't work with Redux DevTools

### DISCUSSION THREAD

**Discussion by NeilTheFisher** [pmndrs/valtio#737] [Bug report]
A workaround is required to show `Map`s on the Redux DevTools, and when the value is changed in the DevTools, it doesn't update the original.

What I did in the reproduction below converts the map to an object so display it's data but this makes it impossible for the DevTools to update the original unless some trickery `subscribe` on the debugState is done, though it wouldn't be hard to implement.

Building this functionality into the `devtools` function if a `Map` or `Set` is used would be great.

It looks like it's possible to have the Redux DevTools serialize maps [this way](https://stackoverflow.com/a/69713337):
```js
const store = Redux.createStore(reducer,
 window.__REDUX_DEVTOOLS_EXTENSION__ &&
 window.__REDUX_DEVTOOLS_EXTENSION__({
  serialize: { 
    options: {
     map: true,
    }
  }
}));
```
If the DevTools can do this, then it can likely also update the Map as well.
Though I'm skeptical about this working since a `proxyMap` is just an emulation of an actual `Map`.

## Link to reproduction

- Open the Redux DevTools with any project that uses `proxyMap` or [this codesandbox](https://codesandbox.io/s/floral-violet-ys8mhk?file=/src/App.tsx)

## Screenshot showing the problem
![image](https://github.com/pmndrs/valtio/assets/47160993/875683c1-3814-467c-a026-fe855aab2507)

**Comment by dai-shi** [comment]
> Though I'm skeptical about this working since a `proxyMap` is just an emulation of an actual `Map`.

Me too.
